gen/defenses/armor_group_generator.py

```diff
@@ -12,36 +12,81 @@
 )
 
 
-def generate_armor_group(faction: str, game, ground_object):
+def generate_armor_group(
+    faction: str, game, ground_object, shorad: bool, player_wants_shorad: bool
+):
     """
     This generate a group of ground units
     :return: Generated group
     """
+    include_shorad: bool = False
+    if shorad == True and player_wants_shorad == True:
+        include_shorad = True
+
     armor_types = (
-        GroundUnitClass.Apc,
         GroundUnitClass.Atgm,
-        GroundUnitClass.Ifv,
         GroundUnitClass.Tank,
+        GroundUnitClass.Apc,
+        GroundUnitClass.Ifv,
     )
     possible_unit = [
         u for u in db.FACTIONS[faction].frontline_units if u.unit_class in armor_types
     ]
     if len(possible_unit) > 0:
         unit_type = random.choice(possible_unit)
-        return generate_armor_group_of_type(game, ground_object, unit_type)
+        return generate_armor_group_of_type(
+            game, ground_object, unit_type, faction, include_shorad
+        )
     return None
 
 
 def generate_armor_group_of_type(
-    game: Game, ground_object: VehicleGroupGroundObject, unit_type: GroundUnitType
+    game: Game,
+    ground_object: VehicleGroupGroundObject,
+    unit_type: GroundUnitType,
+    faction: str,
+    include_shorad: bool,
 ) -> VehicleGroup:
     """
     This generate a group of ground units of given type
     :return: Generated group
     """
     generator = ArmoredGroupGenerator(game, ground_object, unit_type)
     generator.generate()
-    return generator.get_generated_group()
+
+    vehicle_group = generator.get_generated_group()
+
+    if include_shorad == True:
+        shorads = [
+            u
+            for u in db.FACTIONS[faction].frontline_units
+            if u.unit_class == GroundUnitClass.Shorads
+        ]
+        if len(shorads) > 0:
+            shorad = random.choice(shorads)
+            s_posx, s_posy = caluclate_shorad_position(
+                vehicle_group.units[0].position.x,
+                vehicle_group.units[0].position.y,
+                vehicle_group.units[3].position.x,
+                vehicle_group.units[3].position.y,
+            )
+            generator.add_unit(
+                shorad.dcs_unit_type,
+                "group-shorad",
+                s_posx,
+                s_posy,
+                vehicle_group.units[0].heading,
+            )
```

Comment:
This belongs in `ArmoredGroupGenerator`.

Reply:
Ping.

Reply:
splitted: shorad type choice in armor_group_generator, creation in ArmoredGroupGenerator